### PR TITLE
Added filtering to auto-configure version

### PR DESCRIPTION
**Pull request by kaiyaok2**
Using maven filtering from source file to auto-configure version. Avoid needs to manually change versions in future upgrading.

### CODE REVIEW

File: nondex-common/pom.xml [hunk 1]
```diff
@@ -13,6 +13,12 @@
 
   
   <build>
+    <resources>
+       <resource>
+	       <directory>src/main/java/edu/illinois/nondex/common</directory>
```
Comment (darko-marinov) [inline nondex-common/pom.xml:18]: wrong indentation

**Review by darko-marinov — CHANGES_REQUESTED** [review]
Besides that space issue, the rest looks fine if you tested it.

**Comment by kaiyaok2** [comment]
Thanks for pointing that out! I resolved and squashed commits

Best,
Kaiyao

**Comment by kaiyaok2** [comment]
Closing as the changes are addressed in https://github.com/TestingResearchIllinois/NonDex/pull/174

**Pull request closed without merging**